### PR TITLE
feat: remove typeVersions and update aegir to v31

### PR DESCRIPTION
**Pull request by hugomrdias**
- Updates aegir to v31
- Removes typeVersions

BREAKING CHANGE: No longer supports deep type imports only from the entrypoint.

### CODE REVIEW

**Review by vasco-santos — APPROVED** [review]
LGTM

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -14,11 +14,10 @@ jobs:
     - uses: actions/checkout@v2
     - run: npm install
     - run: npx aegir lint
-    - uses: gozala/typescript-error-reporter-action@v1.0.8
```
Comment (Gozala) [inline .github/workflows/main.yml:17]: What the reason for this change ? I personally find inline comments a lot more convenient than reading CI output

**Review by achingbrain — APPROVED** [review]

**Comment by dlecan** [comment]
You can have explained that Node 14 is now required!

**Comment by hugomrdias** [comment]
@dlecan reverted to node 12 in the latest patch release

**Pull request merged**